sunbeam-python/sunbeam/jobs/common.py

```diff
@@ -389,3 +395,36 @@ def str_presenter(dumper: yaml.Dumper, data: str) -> yaml.ScalarNode:
     if data.count("\n") > 0:
         return dumper.represent_scalar("tag:yaml.org,2002:str", data, style="|")
     return dumper.represent_scalar("tag:yaml.org,2002:str", data)
+
+
+def _get_default_no_proxy_settings() -> set:
+    """Return default no proxy settings"""
+    return {
+        "127.0.0.1",
+        "localhost",
+        K8S_CLUSTER_SERVICE_CIDR,
+        K8S_CLUSTER_POD_CIDR,
+        ".svc",
+    }
+
+
+def get_proxy_settings(deployment: Deployment) -> dict:
+    proxy = {}
+    client = deployment.get_client()
+    try:
+        proxy_from_db = read_config(client, PROXY_CONFIG_KEY) or {}
+        proxy = {
+            p.upper(): v
+            for p in ("http_proxy", "https_proxy", "no_proxy")
+            if (v := proxy_from_db.get("proxy", {}).get(p))
+        }
+    except (ClusterServiceUnavailableException, ConfigItemNotFoundException) as e:
+        LOG.debug(f"Using default Proxy settings from provider due to {str(e)}")
+        proxy = deployment.get_default_proxy_settings()
+
+    if "NO_PROXY" in proxy:
+        no_proxy_list = set(proxy.get("NO_PROXY").split(","))
+        missing_no_proxies = _get_default_no_proxy_settings() - no_proxy_list
+        proxy["NO_PROXY"] = ",".join(no_proxy_list.union(missing_no_proxies))
```

Comment:
Computing missing + union is unnecessary, or am I missing the point ?
Set is already taking care of deduplicating
```suggestion
        no_proxies =  _get_default_no_proxy_settings().union(proxy.get("NO_PROXY", "").split(","))
        proxy["NO_PROXY"] = ",".join(no_proxies)
```

Reply:
Done